main.go

```diff
@@ -182,12 +185,64 @@ func wrappedMain() int {
 		}
 	}
 
+	unmanagedProvidersStr := os.Getenv("TF_REATTACH_PROVIDERS")
+	unmanagedProviders := map[addrs.Provider]*plugin.ReattachConfig{}
+	if unmanagedProvidersStr != "" {
+		// {"hashicorp/random":{"Protocol":"grpc","Addr":{"Name":"/tmp/plugin553215095","Net":"unix"},"Pid":19594,"Test":true}}
```

Comment:
This `wrappedMain` is already a pretty busy function, so if it's feasible to factor out this code into a separate function I think that'd help with readability here for future maintainers.

From what I saw in reading through I guess I'm imagining something like this here:

```go
    // The plugin acceptance testing framework can declare it is managing
    // certain providers on Terraform's behalf using this environment variable.
    unmanagedProviders := parseReattachProviders(os.Getenv("TF_REATTACH_PROVIDERS"))`
```